internal/kubeclient/auth.go

```diff
@@ -0,0 +1,28 @@
+package kubeclient
+
+import (
+	"context"
+	"net/http"
+
+	"k8s.io/client-go/rest"
+)
+
+func GetCredential(ctx context.Context, cfg *rest.Config) (string, error) {
+	cfg.Wrap(newAuthorizationHeaderHook)
+	rc, err := rest.HTTPClientFor(cfg)
+	if err != nil {
+		return "", err
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, cfg.Host, nil)
+	if err != nil {
+		return "", err
+	}
+	res, err := rc.Do(req)
+	defer func() {
+		_ = res.Body.Close()
+	}()
+	if err != nil {
+		return "", err
+	}
+	return res.Header.Get(xKargoUserCredentialHeader), nil
```

Comment:
As I understand it, we're inserting some middleware that learns what the token is from the Auth header in the request, then includes that token in a new response header so that we can learn its value here.

I am wondering if that value cannot be obtained more directly from either the `BearerToken` or `BearerTokenFile` fields of the rest config.

Reply:
In a Slack convo, we discussed that this approach will work well for cases involving auth plugins without us having to go to the effort of invoking the plugin ourselves. So the current approach covers more cases with less effort. 👍